python/ray/air/utils/data_batch_conversion_utils.py

```diff
@@ -0,0 +1,88 @@
+from typing import Type
+
+import numpy as np
+import pandas as pd
+
+from ray.air.data_batch_type import DataBatchType
+from ray.air.constants import TENSOR_COLUMN_NAME
+from ray.util.annotations import DeveloperAPI
+
+try:
+    import pyarrow as pa
+except ImportError:
+    pass
+
+
+@DeveloperAPI
+def convert_batch_type_to_pandas(data: DataBatchType) -> pd.DataFrame:
+    """Convert the provided data to a Pandas DataFrame.
+
+    Args:
+        data: Data of type DataBatchType
+
+    Returns:
+        A pandas Dataframe representation of the input data.
+
+    """
+    from ray.air.utils.tensor_extensions.pandas import TensorArray
+
+    if isinstance(data, pd.DataFrame):
+        return data
+
+    elif isinstance(data, np.ndarray):
+        if data.dtype == object:
+            return pd.DataFrame({TENSOR_COLUMN_NAME: list(data)})
+        else:
+            return pd.DataFrame({TENSOR_COLUMN_NAME: TensorArray(data)})
+
+    elif isinstance(data, dict):
+        tensor_dict = {}
+        for k, v in data.items():
+            # Convert numpy arrays to TensorArray.
```

Comment:
Do we raise a good error message if the columns aren't numpy arrays?

Reply:
Yep we should have an instance check on each dict value and raise an error if not an ndarray.

Reply:
Added!